### PR TITLE
[INLONG-6829][Agent] Support for the task delivery of stream_source according to the tag granularity

### LINKED ISSUE #6829: [Feature][Agent] Support for the task delivery of stream_source according to the tag granularity
### Description

Support for the task delivery of stream_source according to the tag granularity

### Use case

Support for the task delivery of stream_source according to the tag granularity

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by thesumery**
### Prepare a Pull Request
- [INLONG-6829][Agent] Support for the task delivery of stream_source according to the tag granularity
- Fixes #6829 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve?*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

**Comment by dockerzhang** [comment]
@thesumery Please add more detail to this PR; thanks. It's better to add a technical solution.

**Comment by dockerzhang** [comment]
@pocozh PTAL, thanks.

File: inlong-manager/manager-test/src/main/resources/h2/apache_inlong_manager.sql [hunk 1]
```diff
@@ -329,6 +330,7 @@ CREATE TABLE IF NOT EXISTS `stream_source`
     `uuid`                varchar(30)           DEFAULT NULL COMMENT 'Mac uuid of the agent running the task',
     `data_node_name`      varchar(128)          DEFAULT NULL COMMENT 'Node name, which links to data_node table',
     `inlong_cluster_name` varchar(128)          DEFAULT NULL COMMENT 'Cluster name of the agent running the task',
+    `inlong_cluster_node_tag` varchar(128)      DEFAULT NULL COMMENT 'Cluster node tag',
     `serialization_type`  varchar(20)           DEFAULT NULL COMMENT 'Serialization type, support: csv, json, canal, avro, etc',
```
Comment (GanfengTan) [inline inlong-manager/manager-test/src/main/resources/h2/apache_inlong_manager.sql:334]: `node_tags` and `inlong_cluster_node_tag` are two fields causing confusion the tag of the cluster.

Comment (thesumery) [reply inlong-manager/manager-test/src/main/resources/h2/apache_inlong_manager.sql:334]: inlong_cluster_node_tag represent this source task must match those node who has these tag.
node_tag represent this node has this tag.

Comment (GanfengTan) [reply inlong-manager/manager-test/src/main/resources/h2/apache_inlong_manager.sql:334]: > inlong_cluster_node_tag represent this source task must match those node who has these tag.
> node_tag represent this node has this tag.

What to do if different cluster name have the same tag?

**Comment by yunqingmoswu** [comment]
Please fix the ut error

File: inlong-manager/manager-test/src/main/resources/h2/apache_inlong_manager.sql (hunk 1 above)
Comment (thesumery) [reply inlong-manager/manager-test/src/main/resources/h2/apache_inlong_manager.sql:334]: this node tag is only to filter task to delivery to which agent node.

Comment (GanfengTan) [reply inlong-manager/manager-test/src/main/resources/h2/apache_inlong_manager.sql:334]: > this node tag is only to filter task to delivery to which agent node.

GET

**Review by GanfengTan — APPROVED** [review]

**Review by pocozh — APPROVED** [review]

File: inlong-agent/agent-common/pom.xml [hunk 2]
```diff
@@ -92,6 +92,11 @@
             <artifactId>commons-io</artifactId>
         </dependency>
 
+        <dependency>
```
Comment (gosonzhang) [inline inlong-agent/agent-common/pom.xml:95]: The liceanse and notice of the agent module need to be supplementedThe liceanse and notice of the agent module need to be supplemented

**Review by gosonzhang — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Review by EMsnap — APPROVED** [review]

**Pull request merged**